### DISCUSSION TITLE
Return an image for a text query and vice-versa?

### DISCUSSION THREAD

**Discussion by HridayM25** [qdrant/qdrant#1501] [General]
Can we get a search return of another type as opposed to the one we queried?
For example, if I search "Messi goal record" as a query, there a lot of "cards" which are images which contain the said records. 
Hence, here if one queries some text, it would make sense to return this said image.
Also, since the "card" will contain information that is better depicted by text, we cannot simply use an image encoder as we will need to do some kind of OCR to derive a semantic meaning.
This was just an idea that crossed my mind.

Example:-
![image](https://user-images.githubusercontent.com/107138441/221571766-f98e33f2-1cf3-4cbe-b9b4-65fd51e8e485.png)

**Comment by monatis** [comment]
Hi @HridayM25, you can make use of cross-encoders such as CLIP  to encode and index image embeddings and then run a query against them with a text embedding.

**Reply by HridayM25** [reply]
Hello,
Thanks for the reply.
Does it make sense to try integrating it directly into our pipeline, so that Qdrant can directly do the above?
